numpy/lib/utils.py

```diff
@@ -105,6 +105,30 @@ def newfunc(*args,**kwds):
         if doc is None:
             doc = depdoc
         else:
+            # Determine the leading whitespace that could be uniformly removed
+            # from the second line onwards. This is based on inspect.cleandoc(),
+            # but avoids the import slowdown incurred by calling on the module.
+            lines = doc.expandtabs().split('\n')
+            margin = sys.maxsize
+            for line in lines[1:]:
+                content = len(line.lstrip())
+                if content:
+                    indent = len(line) - content
+                    margin = min(margin, indent)
+            if margin == sys.maxsize:
+                margin = 0
```

Comment:
nit: could extra lines 112 through 119 as a helper function `_get_indent(lines[1:])`

Reply:
Thanks for the review. I've cleaned up the code and the tests a bit further.